sosia/classes/scientist.py

```diff
@@ -294,10 +304,10 @@ def __init__(self, identifier, year, refresh=False, period=None, eids=None,
             q = f"AU-ID({') OR AU-ID('.join([str(i) for i in identifier])})"
         integrity_fields = ["eid", "author_ids", "coverDate", "source_id"]
         res = base_query("docs", q, refresh, fields=integrity_fields)
-        self._publications = [p for p in res if int(p.coverDate[:4]) <= year]
+        self._publications = [p for p in res if int(p.coverDate[:4]) <= self.year]
```

Comment:
Are you sure about this?

Reply:
Good observation, let's better convert `year` to `int`. I used `self.year` because it it always an `int`.